compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -1274,11 +1476,51 @@ private BLangType createTypeNode(Node type) {
             bLUserDefinedType.pos = getPosition(type);
 
             return bLUserDefinedType;
+        } else if (type.kind() == SyntaxKind.INDEXED_EXPRESSION) {
+            return createBLangArrayFromIndexedExpressionNode((IndexedExpressionNode) type);
         } else {
             return (BLangType) type.apply(this);
         }
     }
 
+    private BLangArrayType createBLangArrayFromIndexedExpressionNode(IndexedExpressionNode indexedExpressionNode) {
+        int dimensions = 1;
+        List<Integer> sizes = new ArrayList<>();
+        do {
+            if (indexedExpressionNode.keyExpression() == null) {
```

Comment:
`indexedExpressionNode.keyExpression()` is called multiple times. extract it to a variable

Reply:
Changed